extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -762,6 +762,9 @@ private void updatePartitionDataFromKafka()
       synchronized (consumerLock) {
         topics = consumer.listTopics(); // updates the consumer's list of partitions from the brokers
       }
+      if (!topics.containsKey(ioConfig.getTopic())) {
+        log.warn("No such topic [%s] found, list of discovered topics [%s]", ioConfig.getTopic(), topics.keySet());
```

Comment:
How about moving this logging to after https://github.com/druid-io/druid/pull/4834/files#diff-6eee87b3aa4eb3a516965fe6e93e25a4R778? The check on topics.containsKey() can be changed to null check on ```partitions```.

Reply:
changed...however I am curious why it is better than this ?

Reply:
Thanks. It's very trivial, but I think it's better to read because the existence check on topic is done in one place.